apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx

```diff
@@ -133,30 +141,95 @@ const getNativeElementsList = (elements: SBEnumType['value']): JSX.Element => {
   );
 };
 
-const RenderArgsTable = ({ hideArgsTable, primaryStory }: { primaryStory: PrimaryStory; hideArgsTable: boolean }) => {
+const slotRegex = /as\?:\s*"([^"]+)"/;
+
+function withSlotEnhancer(story: PreparedStory) {
+  const updatedArgTypes = { ...story.argTypes };
+
+  type InternalComponentApi = {
+    __docgenInfo: { props?: Record<string, { type: { name: string } }> };
+    [k: string]: unknown;
+  };
+  const component = story.component as InternalComponentApi;
+  const docGenProps = component?.__docgenInfo?.props;
+
+  if (!docGenProps) {
+    return component;
+  }
+
+  Object.entries(docGenProps).forEach(([key, argType]) => {
+    const value: string = argType?.type?.name;
+    if (value.includes('WithSlotShorthandValue')) {
+      const match = value.match(slotRegex);
+      if (match) {
+        component.__docgenInfo.props![key].type.name = `Slot<\"${match[1]}\">`;
+        // @ts-expect-error - storybook doesn't ship proper types (value is missing)
+        updatedArgTypes[key].type.value = `Slot<\"${match[1]}\">`;
+      } else {
+        component.__docgenInfo.props![key].type.name = `Slot`;
+        // @ts-expect-error - storybook doesn't ship proper types (value is missing)
+        updatedArgTypes[key].type.value = `Slot`;
+      }
+    }
+  });
+
+  return component;
+}
+const RenderArgsTable = ({
+  hideArgsTable,
+  story,
+  argTypes,
+}: {
+  story: PrimaryStory;
+  hideArgsTable: boolean;
+  argTypes: StrictArgTypes;
+}) => {
   const styles = useStyles();
+
+  const storyCopy = { ...story };
+  const { component } = withSlotEnhancer(storyCopy);
+
+  // const hasSlot = story.argTypes.as && story.argTypes.type.value === 'Slot';
   return hideArgsTable ? null : (
     <>
-      <ArgsTable of={primaryStory.component} />
-      {primaryStory.argTypes.as && primaryStory.argTypes.as?.type?.name === 'enum' && (
-        <div className={styles.nativeProps}>
-          <InfoFilled className={styles.nativePropsIcon} />
-          <div className={styles.nativePropsMessage}>
-            <b>
-              Native props are supported <span role="presentation">🙌</span>
-            </b>
-            <span>
-              All HTML attributes native to the {getNativeElementsList(primaryStory.argTypes.as.type.value)}, including
-              all <code>aria-*</code> and <code>data-*</code> attributes, can be applied as native props on this
-              component.
-            </span>
+      {story.argTypes.as && story.argTypes.as?.type?.name === 'enum' && (
```

Comment:
we have here a repetition already - time to encapsulate this into separate component

**Task:**

- [ ] create a local component with following API usage:

```tsx
<AdditionalApiDocs>
 custom content
</AdditionalApiDocs>
```

- [ ] render these components with extra info conditionaly


```tsx
const hasArgAsProp = story.argTypes.as && story.argTypes.as?.type?.name === 'enum';
const hasArgSlotProp =  ???;

return 
{ hasArgAsProp ? <AdditionalApiDocs>
 custom content
</AdditionalApiDocs>: null }

{ hasArgSlotProp ? <AdditionalApiDocs>
 custom content
</AdditionalApiDocs>: null }
```